swap/src/bin/asb.rs

```diff
@@ -228,17 +233,39 @@ pub async fn main() -> Result<()> {
         }
         Command::History => {
             let db = open_db(config.data.dir.join("sqlite"), AccessMode::ReadOnly, None).await?;
-
             let mut table = Table::new();
 
-            table.set_header(vec!["SWAP ID", "STATE"]);
-
-            for (swap_id, state) in db.all().await? {
-                let state: AliceState = state.try_into()?;
-                table.add_row(vec![swap_id.to_string(), state.to_string()]);
+            table.set_header(vec![
+                "Swap ID",
+                "Start Date",
+                "State",
+                "Bitcoin Lock TxId",
+                "BTC Amount",
+                "XMR Amount",
+                "Exchange Rate",
+                "Trading Partner Peer ID",
+                "Completed",
+            ]);
+
+            let all_swaps = db.all().await?;
+            for (swap_id, state) in all_swaps {
+                match SwapDetails::from_db_state(swap_id.clone(), state, &db).await {
```

Comment:
We're calling from_db_state for every swap, but from_db_state iterates through all swaps itself. That's O(n^2), so we should probably move that into a single database query

Reply:
`from_db_state` does not iterate through all swaps. It iterates through all saved states of a particular swap.